frame/nomination-pools/src/lib.rs

```diff
@@ -1943,25 +1947,108 @@ pub mod pallet {
 			Ok(())
 		}
 
-		/*
+		
+		// TODO: Benchmark?
+		/// Doesn't mean that the poolId has been removed from the BondedPools, just means that there is no active depositor and no stake in the pool
 		/// Create a new delegation pool with a previously used pool id
 		/// 
 		/// # Arguments
 		/// 
 		/// same as `create` with the inclusion of
 		/// * `pool_id` - `Option<PoolId>`, if `None` this is similar to `create`.
 		///    if `Some(claim)` the caller is claiming that `claim` A.K.A PoolId is not in use. 
-		pub fn create_with_pool_id() {
+		#[pallet::weight(T::WeightInfo::create())]
+		#[transactional]
+		pub fn create_with_pool_id(
 			origin: OriginFor<T>,
 			#[pallet::compact] amount: BalanceOf<T>,
 			root: AccountIdLookupOf<T>,
 			nominator: AccountIdLookupOf<T>,
 			state_toggler: AccountIdLookupOf<T>,
-			pool_id: Option<PoolId>,
-		} -> DispatchResult {
-			
+			state_claim: Option<PoolId>,
+		 ) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let root = T::Lookup::lookup(root)?;
+			let nominator = T::Lookup::lookup(nominator)?;
```

Comment:
YOu should make a helper function, not duplicate the whole code :(

Reply:
Right!